src/Internal/Service/OrderService.php

```diff
@@ -254,6 +254,87 @@ public function attach_exchange_info_to_order( int $order_id, WC_Payments_API_Ch
 		$order->save_meta_data();
 	}
 
+	/**
+	 * Gets currently attached intent ID of the order.
+	 *
+	 * @param  int $order_id Order ID.
+	 *
+	 * @return string|null Intent ID for the order. Null if no intent ID attached to order.
+	 * @throws Order_Not_Found_Exception
+	 */
+	public function get_intent_id( int $order_id ): ?string {
+		$order     = $this->get_order( $order_id );
+		$intent_id = (string) $order->get_meta( '_intent_id', true );
+		if ( empty( $intent_id ) ) {
+			return null;
+		}
+		return $intent_id;
+	}
+
+	/**
+	 * Updates the order status from the given intent object.
+	 *
+	 * @param  int                                $order_id Order ID.
+	 * @param  WC_Payments_API_Abstract_Intention $intent Intent object.
+	 *
+	 * @return void
+	 * @throws Order_Not_Found_Exception
+	 */
+	public function update_order_status_from_intent( int $order_id, WC_Payments_API_Abstract_Intention $intent ): void {
+		$this->legacy_service->update_order_status_from_intent( $this->get_order( $order_id ), $intent );
+	}
+
+	/**
+	 * Gets cart hash for the given order ID.
+	 *
+	 * @param  int $order_id  ID of the order.
+	 *
+	 * @return string Cart hash for the order.
+	 * @throws Order_Not_Found_Exception
+	 */
+	public function get_cart_hash( int $order_id ): string {
+		$order = $this->get_order( $order_id );
+		return $order->get_cart_hash();
+	}
+
+	/**
+	 * Checks if the order has one of paid statuses.
+	 *
+	 * @param  int $order_id  ID of the order.
+	 *
+	 * @return bool True if the order has one of paid statuses, false otherwise.
+	 * @throws Order_Not_Found_Exception
+	 */
+	public function is_paid( int $order_id ): bool {
+		return $this->get_order( $order_id )->has_status( wc_get_is_paid_statuses() );
```

Comment:
`LegacyProxy`` needed here.

Reply:
Not sure why we also need `LegacyProxy` here while `$this->get_order()` is actually using `LegacyProxy`?

Reply:
Oh, sorry, I missed `wc_get_is_paid_statuses`. Going to fix it now.

Reply:
01e204ce9d4671655cec53453ddbb069898f58f6 would fix it.